cmake/get_info.cmake

```diff
@@ -0,0 +1,310 @@
+SET(detailed_log "${CMAKE_BINARY_DIR}/detailed.log")
+SET(minimal_log  "${CMAKE_BINARY_DIR}/minimal.log")
+FILE(REMOVE ${detailed_log} ${minimal_log})
+
+MACRO(_both)
+  # Write to both log files:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_detailed)
+  # Only write to detailed.log:
+  FILE(APPEND ${detailed_log} "${ARGN}")
+ENDMACRO()
+
+MACRO(_minimal)
+  # Only write to minimal.log:
+  FILE(APPEND ${minimal_log} "${ARGN}")
+ENDMACRO()
+
+FUNCTION(build_type_spec log_type var_name)
```

Comment:
question: Do we want to deal with other build types (sanitizers)?

Reply:
Regarding this, I am guessing you mean printing out a log when sanitizers are being built. I see this as some quick information for the user when they configure Ginkgo. I am not sure how many actually want to run Ginkgo with sanitizers.

Reply:
Fair point, I would only like to avoid anything breaking when we build other build types ;)